### PR TITLE
Replicate _users database to table couchdb_medic_users

### PR DESCRIPTION
**Pull request by kennsippell**
#137 

Change to `couch2pg@0.7.1` is https://github.com/medic/couch2pg/pull/45

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,13 +17,13 @@
   "dependencies": {
     "chalk": "2.4.1",
     "clui": "0.3.6",
-    "couch2pg": "^0.7.0",
+    "couch2pg": "^0.7.1",
```
Comment (kennsippell) [inline package.json:20]: Published result of https://github.com/medic/couch2pg/pull/45

File: package.json [hunk 2]
```diff
@@ -17,13 +17,13 @@
...
     "figlet": "1.2.0",
     "inquirer": "^3.0.6",
     "loglevel": "^1.4.1",
     "pg": "^8.0.0",
     "pg-format": "1.0.4",
-    "pg-promise": "^10.0.0",
+    "pg-promise": "^10.11.1",
```
Comment (kennsippell) [inline package.json:26]: Running `npm i` bumped `pg-promise` to 10.15.x which dropped support for NodeJS<12. This causes CI tests to fail on Node 8 and 10. I've proposed to around around the issue by pinning the version to the current production version.

File: package.json [hunk 3]
```diff
@@ -17,10 +17,11 @@
   "dependencies": {
     "chalk": "2.4.1",
     "clui": "0.3.6",
-    "couch2pg": "^0.7.0",
+    "couch2pg": "^0.7.1",
     "figlet": "1.2.0",
     "inquirer": "^3.0.6",
     "loglevel": "^1.4.1",
+    "loglevel-message-prefix": "^3.0.0",
```
Comment (kennsippell) [inline package.json:24]: This is used directly in `libs/medic/log.js` so we should include it as a dependency.

**Comment by kennsippell** [comment]
@njogz You might think you've seen this exact PR because of #136. Truly very similar! 
I appreciate your eyes here again.

File: env.js [hunk 4]
```diff
@@ -23,5 +23,6 @@ module.exports = function() {
     syncSentinelDb: notFalse(process.env.SYNC_DB_SENTINEL),
     syncUserMetaDb: notFalse(process.env.SYNC_DB_USER_META),
     syncLogsDb: notFalse(process.env.SYNC_DB_LOGS),
+    syncUsersDb: process.env.SYNC_DB_USERS === 'true',
```
Comment (kennsippell) [inline env.js:26]: Should this still default to off now that security information is removed from the docs?

Comment (mrjones-plip) [reply env.js:26]: @kennsippell  - now that we cleanse the hashes before inserting to postgres, I'm OK if we default to being on.  I definitely can see the value of having user data side by side with the rest of the lot.  

Thanks for checking!

Comment (mrjones-plip) [reply env.js:26]: @kennsippell  - now that we cleanse the hashes before inserting to postgres, I'm OK if we default to being on.  I definitely can see the value of having user data side by side with the rest of the lot.  

Thanks for checking!

**Comment by yrimal** [comment]
Are there any materialized views that you think are useful to have across all deployments? For example, we create multiple views and materialized views during initial setup based on which file is being run [here](https://github.com/medic/cht-couch2pg/tree/main/libs/analytics/migrations). It will be nice to have those views/materialized views created and executed as a part of c2pg rather than requiring us to recreate same objects later.

**Comment by kennsippell** [comment]
Thanks for the feedback @yrimal. I personally find the _user data to be quite simple. The table is usually small (<12000 users max for CHT instances). We might consider a reusable view, but I'm not sure it adds that much value. Would you use it and prefer to have it?

File: libs/medic/runner.js [hunk 5]
```diff
@@ -35,6 +35,11 @@ const replicateAll = async (couchUrl, pgconn, opts) => {
     if (opts.syncLogsDb) {
       await replicateDatabase(`${couchUrl}-logs`, 'couchdb_medic_logs');
     }
+
+    if (opts.syncUsersDb) {
+      const couchUrlWithoutPath = couchUrl.substring(0, couchUrl.lastIndexOf('/'));
```
Comment (dianabarsan) [inline libs/medic/runner.js:40]: I dislike all url manipulation in this file. I think the URL never even gets normalized. Not something that needs to be done in this PR though.

**Review by dianabarsan — COMMENTED** [review]
Just nits inline. Thanks for adding this!

**Review by dianabarsan — APPROVED** [review]

**Pull request merged**